core/arch/arm/plat-marvell/platform_config.h

```diff
@@ -0,0 +1,138 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef PLATFORM_CONFIG_H
+#define PLATFORM_CONFIG_H
+
+#include <stdint.h>
+
+/* Make stacks aligned to data cache line length */
+#define STACK_ALIGNMENT		64
+
+#ifdef ARM64
+#ifdef CFG_WITH_PAGER
+#error "Pager not supported yet"
+#endif
+#else
+#error "32 bit mode not supported yet"
+#endif /*ARM64*/
+
+/* SDP enable but no pool defined: reserve 4MB for SDP tests */
+#if defined(CFG_SECURE_DATA_PATH) && !defined(CFG_TEE_SDP_MEM_BASE)
+#define CFG_TEE_SDP_MEM_TEST_SIZE	0x00400000
+#else
+#define CFG_TEE_SDP_MEM_TEST_SIZE	0
+#endif
+
+#if defined(PLATFORM_FLAVOR_armada7k8k)
+/*
+ * armada7k8k specifics.
+ */
+#define TEE_RES_CFG_8M
+
+#define MVEBU_REGS_BASE		0xF0000000
+
+/* GICv2 */
+#define MVEBU_GICD_BASE		0x210000
+#define MVEBU_GICC_BASE		0x220000
+#define GIC_DIST_BASE			(MVEBU_REGS_BASE + MVEBU_GICD_BASE)
+#define GIC_CPU_BASE			(MVEBU_REGS_BASE + MVEBU_GICC_BASE)
+
+#define GIC_BASE		GIC_DIST_BASE
+
+/* UART */
+#define PLAT_MARVELL_BOOT_UART_BASE		(MVEBU_REGS_BASE + 0x512000)
+#define PLAT_MARVELL_BOOT_UART_CLK_IN_HZ        200000000
+#define MARVELL_CONSOLE_BAUDRATE                            115200
+
+#define CONSOLE_UART_BASE	PLAT_MARVELL_BOOT_UART_BASE
+
+/* Location of trusted dram */
+#define TZDRAM_BASE		0x04400000
+#define TZDRAM_SIZE		0x00C00000
+
+#define CFG_TEE_CORE_NB_CORE	4
+
+#define CFG_SHMEM_START		(TZDRAM_BASE + TZDRAM_SIZE)
+#define CFG_SHMEM_SIZE		0x00400000
+
+#define GICC_OFFSET		0x10000
+#define GICD_OFFSET		0x0
+
+/* MCU */
+#define MCU_BASE			0xF0020000
+#define MCU_REG_SIZE		0x1000
+#define MC_SCR_REGISTER	0xF06F0204
+#define MC_SCR_REG_SIZE	0x1000
+
+#else
+#error "Unknown platform flavor"
+#endif
+
+#define CFG_TEE_RAM_VA_SIZE	(4 * 1024 * 1024)
```

Comment:
What about using `SIZE_4M` here ? 
You could consider using `SIZE_4K` instead of the 2 `#define xxx_SIZE 0x1000` above.

Reply:
I could not use SIZE_4M for CFG_TEE_RAM_VA_SIZE. Looks like kern.ls do not recognize "UL"

Reply:
Try to use `UINTPTR_C()` instead of "UL"

Reply:
It's not only caused by "UL", but also the different definitions of ROUND_UP in util.h and kernel/kern.ld.S.